spanner/spanner_samples.rb

```diff
@@ -1495,6 +1505,211 @@ def write_read_float64_array project_id:, instance_id:, database_id:
+    end
+  end
+  # [END spanner_list_backup_operations]
+end
+
+def list_database_operations project_id:, instance_id:
+  # [START spanner_list_database_operations]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  jobs = instance.database_operations filter: "metadata.@type:RestoreDatabaseMetadata"
+  jobs.each do |job|
+    if job.error?
+      puts job.error
+    else
+      restore_info = job.database.restore_info
+      puts "Database #{job.database.database_id} restored from backup #{restore_info.backup_info.backup_id} is in #{job.database.state} state"
+    end
+  end
+  # [END spanner_list_database_operations]
+end
+
+def list_backups project_id:, instance_id:
+  # [START spanner_list_backups]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  puts "All backups"
+  instance.backups.all.each do |backup|
+    puts backup.backup_id
+  end
+
+  puts "All backups that contains a name \"test_bu_\":"
+  instance.backups(filter: "name:test_bu_").all.each do |backup|
+    puts backup.backup_id
+  end
+
+  puts "All backups for a database that contains a name \"test_db_\":"
+  instance.backups(filter: "database:test_db_").all.each do |backup|
+    puts backup.backup_id
+  end
+
+  puts "All backups that expire before a timestamp:"
+  instance.backups(filter: "expire_time < \"#{(Time.now + 315_360_000).iso8601}\"").all.each do |backup|
+    puts backup.backup_id
+  end
+
+  puts "All backups with a size greater than 500 bytes:"
+  instance.backups(filter: "size_bytes > 500").all.each do |backup|
+    puts backup.backup_id
+  end
+
+  puts "All backups that were created after a timestamp that are also ready:"
+  instance.backups(filter: "create_time >= \"#{(Time.now - 10_000).iso8601}\" AND state:READY").all.each do |backup|
+    puts backup.backup_id
+  end
+
+  puts "All backups with pagination:"
+  list = instance.backups page_size: 5
+  list.each do |backup|
+    puts backup.backup_id
+  end
+  # [END spanner_list_backups]
+end
+
+def delete_backup project_id:, instance_id:, backup_id:
+  # [START spanner_delete_backup]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # backup_id = "Your Spanner backup ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  backup = instance.backup backup_id
+  backup.delete
+  puts "Backup #{backup_id} deleted"
+  # [END spanner_delete_backup]
+end
+
+def update_backup_expiration_time project_id:, instance_id:, backup_id:
```

Comment:
```suggestion
def update_backup project_id:, instance_id:, backup_id:
```

The reason being that we may end up adding other parameters to update later on.